### PR TITLE
[ENHANCEMENT] `argilla`: align usage of questions

### LINKED ISSUE #5231: [FEATURE] Access questions by name
In general, settings can be accessed by name. So settings.fields["text"] or settings.vectors["embeddings"] are valid. I realized this is not possible for questions, which must be iterated to be accessed.
````
for q in settings.questions:
   if q.name == "...":
      ...
````
Questions should be aligned with the rest of the settings.

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR reviews and aligns the questions' usage with the rest of the settings. Now, questions can be accessed as the rest of the settings:
```python
settings.questions["name"]
```

Note: Updating questions for existing datasets is still not possible. But we can add questions to settings before creating the dataset:
```python
settings.questions.add(...)
dataset = rg.Dataset(..., settings=settings).create()
```

Closes https://github.com/argilla-io/argilla/issues/5231

**Type of change**

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (change restructuring the codebase without changing functionality)
- Improvement (change adding some improvement to an existing functionality)
- Documentation update

**How Has This Been Tested**

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/src/argilla/settings/_resource.py [hunk 1]
```diff
@@ -386,10 +360,15 @@ def __init__(self, settings: "Settings", properties: List[Property]):
         for property in properties or []:
             self.add(property)
 
-    def __getitem__(self, key: Union[str, int]) -> Optional[Property]:
+    def __getitem__(self, key: Union[UUID, str, int]) -> Optional[Property]:
         if isinstance(key, int):
             return list(self._properties_by_name.values())[key]
-        return self._properties_by_name.get(key)
+        if isinstance(key, UUID):
```
Comment (davidberenstein1957) [inline argilla/src/argilla/settings/_resource.py]:
```suggestion
        elif isinstance(key, UUID):
```

File: argilla/tests/unit/test_settings/test_settings.py [hunk 2]
```diff
@@ -93,6 +96,34 @@ def test_settings_validation_with_duplicated_names(self):
             settings.validate()
 
 
+class TestSettingsProperties:
```
Comment (davidberenstein1957) [inline argilla/tests/unit/test_settings/test_settings.py]: should be try t test access by `int` /index too?

**Review by davidberenstein1957 — APPROVED** [review]
LFTM, perhaps we should double check the docs?

**Comment by frascuchon** [comment]
> LFTM, perhaps we should double check the docs?

@nataliaElv is already working on this [here](https://github.com/argilla-io/argilla/pull/5254)

**Pull request merged**